api/main.py

```diff
@@ -655,3 +664,14 @@ async def put_regression(regression_id: str, regression: Regression,
         sub_app.app.add_exception_handler(
             errors.InvalidId, invalid_id_exception_handler
         )
+
+
+@app.middleware("http")
+async def redirect_http_requests(request: Request, call_next):
+    """Redirect request with version prefix when no version is provided"""
+    path = request.scope['path']
+    prefix = path.split('/')[1]
+    if prefix not in API_VERSIONS:
+        request.scope['path'] = '/latest' + path
```

Comment:
As a side effect, looking for a non-existing version will implicitly redirect to `latest`.  If we only have a single-digit version number e.g. `v0`, `v1` etc. then it's probably fine, even though it might be a bit unexpected.  If we do however introduce dotted versions then this might lead to some intricate issues.  For example:

`http://host/v123/hello` -> `http://host/latest/hello` would be better with an error instead
`http://host/v0.2` -> `http://host/latest/hello`  would be very misleading if the latest version was in fact `v1` for example

So maybe a simple thing to avoid this is to add a check if `prefix` matches a regex e.g. `'^v\d+'` and return an error if that's not a known API version.